drivers/network/tcpip/ip/transport/tcp/accept.c

```diff
@@ -77,10 +77,10 @@ NTSTATUS TCPListen(PCONNECTION_ENDPOINT Connection, UINT Backlog)
             if (NT_SUCCESS(Status))
             {
                 /* Allocate the port in the port bitmap */
-                Connection->AddressFile->Port = TCPAllocatePort(LocalAddress.Address[0].Address[0].sin_port);
-
-                /* This should never fail */
-                ASSERT(Connection->AddressFile->Port != 0xFFFF);
+                UINT AllocatedPort = TCPAllocatePort(LocalAddress.Address[0].Address[0].sin_port);
+                /* This should never fail unless all ports are in use */
+                ASSERT(AllocatedPort != (UINT)-1);
```

Comment:
Is the assert needed here ? Maybe the function should only return a status code to usermode (exhausting all local ports can happen in expected system lifecycle, for instance if server is target of DDoS or runs poorly-coded usermode application)

Reply:
This was a request from @learn-more in the previous PR, but based on the assumption that "this should never fail" which was the code comment.
But you're right, the error case is correctly manage and this CAN happen under very loaded use cases (or poorly written networking app) so the best option is NOT to ASSERT but to fail gracefully (and i'd suggest to log an ERR since it should be very unlikely)

Reply:
Latest commit attempts to address this issue. Please review and approve if appropriate.